Classes/Utility/UserUtility.php

```diff
@@ -246,18 +246,22 @@ public static function getDirtyPropertiesFromUser(User $changedObject)
                 && !in_array($propertyName, $ignoreProperties)
             ) {
                 $newPropertyValue = $changedObject->{'get' . ucfirst($propertyName)}();
-                if (!is_object($oldPropertyValue) || !is_object($newPropertyValue)) {
+                if (!is_object($oldPropertyValue) && !is_object($newPropertyValue)) {
                     if ($oldPropertyValue !== $newPropertyValue) {
                         $dirtyProperties[$propertyName]['old'] = $oldPropertyValue;
                         $dirtyProperties[$propertyName]['new'] = $newPropertyValue;
                     }
                 } else {
-                    if (get_class($oldPropertyValue) === 'DateTime') {
+                    if (($oldPropertyValue != null && get_class($oldPropertyValue) === 'DateTime') || ($newPropertyValue != null && get_class($newPropertyValue) === 'DateTime')) {
                         /** @var $oldPropertyValue \DateTime */
                         /** @var $newPropertyValue \DateTime */
-                        if ($oldPropertyValue->getTimestamp() !== $newPropertyValue->getTimestamp()) {
-                            $dirtyProperties[$propertyName]['old'] = $oldPropertyValue->getTimestamp();
-                            $dirtyProperties[$propertyName]['new'] = $newPropertyValue->getTimestamp();
+
+                        $oldTimestamp = $oldPropertyValue != null ? $oldPropertyValue->getTimestamp() : 0;
```

Comment:
`$oldPropertyValue` cannot be null and must therefore be a `DateTime` object, it's checked in the condition before.

Reply:
The condition in line 255 only checks if at least one of the two propertyValues is a DateTime. Therefore either one of them could still be `null` inside the condition and has to checked again,